src/engine.ts

```diff
@@ -449,16 +472,69 @@ export default class Engine {
     return spaces;
   }
 
-  possibleBoardActions(player: PlayerEnum): BoardAction[] {
-    return  Object.values(BoardAction).filter(pwract => this.boardActions[pwract] && this.player(player).canPay(Reward.parse(boardActions[pwract].cost)));
+  possibleBoardActions(player: PlayerEnum) {
+    const commands = [];
+
+    let poweracts = Object.values(BoardAction).filter(pwract => this.boardActions[pwract] && this.player(player).canPay(Reward.parse(boardActions[pwract].cost)));
+    if (poweracts.length > 0) {
+      commands.push({
+        name: Command.Action,
+        player,
+        data: { poweracts: poweracts.map(act => ({
+          name: act,
+          cost: boardActions[act].cost,
+          income: boardActions[act].income
+        }))}
+      });
+    };
+
+    return commands;
+
+  }
+
+  possibleFreeActions(player: PlayerEnum) {
+
+    // free action - spend
+    const acts = [];
+    const commands = [];
+    for (let i = 0; i < freeActions.length; i++) {
+      if (this.player(player).canPay(Reward.parse(freeActions[i].cost))) {
+        acts.push({ 
+          cost: freeActions[i].cost,
+          income: freeActions[i].income  
+        });
+      };
+    };
+
+    if (acts.length > 0) {
+      commands.push({
+        name: Command.Spend,
+        player,
+        data: { acts }
+      });
+    }
+
+    //free action - burn
+    //TODO generate burn actions based on  Math.ceil( engine.player(player).data.power.area2 / 2)
+    if (this.player(player).data.power.area2 >= 2) {
+      commands.push({
+        name: Command.BurnPower,
+        player,
+        data: _.range(1, Math.floor(this.player(player).data.power.area2 / 2) + 1)
+      });
+    }
+    return commands;
+
   }
 
   /** Next player to make a move, after current player makes their move */
   moveToNextPlayer(command: Command): PlayerEnum {
+    const playerPos = this.turnOrder.indexOf(this.currentPlayer);
     const subPhaseTurn = this.roundSubCommands.length > 0;
     
     if (subPhaseTurn) {
-      // This is a sub command, wait until they are all done
+      // This is a sub command, sets currentplayer 
+      this.currentPlayer = this.roundSubCommands[0].player;
```

Comment:
I'm a bit worried about this (why I added `this.playerToMove()` in a recent commit), since the next player is determined based on the current player. You seem to have mitigated that by setting `this.nextPlayer`). I hope it all works fine with games with 3 / 4 players!

Reply:
I know. I'm not fully convinced neither. I'll check it again later. We will have probably to end up with a proper action workflow management.

Reply:
BTW, the only code which is changing regular turnOrder is the leeching phase. So far so good.

Reply:
Actually this line doesn't seem to be needed. I pulled your changes and removed it, and the tests are still fine

Reply:
Let me check, I'm not sure. I think it's still working because most of the tests are only two player games.

Reply:
You right. It works because of ```this.playerToMove``` which is considering availableCommands ! Nice!